testonly/integration/mapenv.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"database/sql"
+	"sync"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/keys"
+	"github.com/google/trillian/extension"
+	"github.com/google/trillian/quota"
+	"github.com/google/trillian/server"
+	"github.com/google/trillian/storage/mysql"
+	stestonly "github.com/google/trillian/storage/testonly"
+	"google.golang.org/grpc"
+)
+
+// MapEnv is a map server and connected client.
+type MapEnv struct {
+	registry     extension.Registry
```

Comment:
Maybe.. my thought is that by keeping them private, we keep a bias towards data encapsulation. We can make them public one by one as we need them on a case by case basis.  Maintenance of the Env struct is easier the fewer things there are poking around with it.